docs/rfcs/003-open-in-atom.md

```diff
@@ -0,0 +1,137 @@
+
+
+### Merge conflict
+
+1. Clones the repo (if needed)
+2. Checks out the branch
+3. Opens a new Atom window
+    - Or focuses the window if the repo already exists as an opened project
+4. Opens the Git panel (showing all conflicting files)
+5. Opens the (first) conflicting file
+6. Scrolls to the (first) conflict
+
+![image](https://user-images.githubusercontent.com/378023/41271078-6c8e6a6c-6e49-11e8-96ee-5b0b5e55ef3d.png)
+
+### File
+
+1. Clones the repo (if needed)
+2. Checks out the branch
+3. Opens a new Atom window
+    - Or focuses the window if the repo already exists as an opened project
+4. Opens the file
+
+![image](https://user-images.githubusercontent.com/378023/41266930-30a24092-6e34-11e8-9c6e-36af97bc2bb3.png)
+
+### Line
+
+1. Clones the repo (if needed)
+2. Checks out the branch
+3. Opens a new Atom window
+    - Or focuses the window if the repo already exists as an opened project
+4. Opens the file
+5. Scrolls to the line
+6. Shows a blinking cursor at the beginning of the line 
+
+![open-in-atom](https://user-images.githubusercontent.com/378023/41338002-ccfe8944-6f2b-11e8-95d7-7e823e6d8643.png)
+
+
+## Drawbacks
+
+None that come to mind.
+
+## Rationale and alternatives
+
+By reducing the friction for moving from github.com to Atom, we can improve the following:
+
+- Higher quality of code reviews.
+    - Sometimes a reviewer might have the urge to suggest something, but they are not sure without testing it first. But since testing locally takes so many steps, they might just leave a comment like "Looks good to me" and move on. 👉 The easier it is to test changes, the better the reviews will become.
+- More contributions.
+    - If a reviewer tests something locally and makes edits, they might commit the changes and push it back to the PR. 👉 Reviewers that can easily edit and test a PR more likely become collaborators.
+- Better integration with other applications.
+    - Be able to open files, lines, merge conflicts etc. in Atom isn't just limited to github.com, it's also possible for other applications. For example if there is a merge conflict in GitHub Desktop, instead of manually switching and trying to find the conflict, GitHub Desktop could send you to Atom in a more seamless way.
+
+### Alternatives
+
+An alternative is to let users discover PRs/issues or get notified about changes **within** Atom. This would certainly be nice to have, but it will be a very long road until a "never have to leave Atom" is reached. Especially for our tiny team. We also don't have to pick one or the other and both approaches can complement each other.
+
+An alternative to have an Atom integration directly build into github.com would be to create a **browser extension**. A benefit is that it can be extended to other web sites/apps. For example an "Open in Atom" link inside Gmail for GitHub notifications. Downside: Users have to first install the extension.
```

Comment:
Yah, I much prefer this to writing browser extensions, I think - much more discoverable.

Reply:
This functionality could be easily implemented as a bookmarklet.

![image](https://user-images.githubusercontent.com/11719160/41985254-b3bda820-7a2a-11e8-989e-41fbf19238d0.png)

For example the following pseudo-link would send any URL to the `git-client` protocol (which is used by Visual Studio):
```
javascript:window.location="git-client://clone?repo="+document.URL
```

This might be useful for prototyping or for users who want this functionality close at hand. Just a crazy idea. 😉